### PR TITLE
[ECO-4982] Implement `Messages` discontinuities (CHA-M7)

### LINKED ISSUE #47: Integrate room lifecycle manager with the rest of the SDK
Split from #28.

This means:

- [x] triggering the specified room lifecycle operations from elsewhere in the SDK (presumably `Room`, or in the case of `RELEASE`, `Rooms`)
- [x] bridging the underlying realtime channels to the `RoomLifecycleContributorChannel` protocol
- [x] implementing CHA-RL3h, which says to release the underlying Realtime channel after the `RELEASE` operation, and CHA-RL1d, which says to discard the room instance
- [x] integrating discontinuity events (e.g. CHA-M7)
- [x] passing contributors into other features (i.e. only fetch channels once)
- [x] update integration tests for RELEASE

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4982) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
**Note: This PR is based on top of #107; please review that one first.**

Based on same spec as 8daa191.

Resolves #47.

## Release Notes

- **New Features**
	- Introduced `FeatureChannel` protocol and `DefaultFeatureChannel` struct for enhanced channel functionality.
	- Updated `DefaultMessages` class to utilize `featureChannel`, improving message handling and subscription management.
	- Added `subscribeToDiscontinuities` method in `DefaultRoomLifecycleContributor` for better discontinuity event handling.

- **Bug Fixes**
	- Enhanced error handling in message subscriptions to ensure proper validation of incoming messages.

- **Tests**
	- Updated tests for `DefaultMessages` to reflect changes in initialization and added scenarios for testing discontinuities.
	- Introduced `MockFeatureChannel` for simulating channel behavior in tests.

### CODE REVIEW

**Review by maratal — APPROVED** [review]
LGTM, left minor suggestion.

File: Tests/AblyChatTests/DefaultMessagesTests.swift [hunk 1]
```diff
@@ -65,4 +68,24 @@ struct DefaultMessagesTests {
...
+        let featureChannel = MockFeatureChannel(channel: channel)
+        let messages = await DefaultMessages(featureChannel: featureChannel, chatAPI: chatAPI, roomID: "basketball", clientID: "clientId")
+
+        // When: The feature channel emits a discontinuity through `subscribeToDiscontinuities`
+        let featureChannelDiscontinuity = ARTErrorInfo.createUnknownError() // arbitrary
+        let messagesDiscontinuitySubscription = await messages.subscribeToDiscontinuities()
+        await featureChannel.emitDiscontinuity(featureChannelDiscontinuity)
+
+        // Then: The DefaultMessages instance emits this discontinuity through `subscribeToDiscontinuities`
+        let messagesDiscontinuity = try #require(await messagesDiscontinuitySubscription.first { _ in true })
```
[the remark this answers is not in the document]
Comment (maratal) [reply Tests/AblyChatTests/DefaultMessagesTests.swift:88]: Maybe `first()` function then?

[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Tests/AblyChatTests/DefaultMessagesTests.swift:88]: You mean just for `Subscription`?

[the remark this answers is not in the document]
Comment (maratal) [reply Tests/AblyChatTests/DefaultMessagesTests.swift:88]: Or the whole `AsyncSequence`?

[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply Tests/AblyChatTests/DefaultMessagesTests.swift:88]: Yeah, that causes the throwing issue again, though:

image

I don't really want to spend more time looking into this, can we just leave as is?

[the remark this answers is not in the document]
Comment (maratal) [reply Tests/AblyChatTests/DefaultMessagesTests.swift:88]: Sure, not that important, resolving this.

**Pull request merged**